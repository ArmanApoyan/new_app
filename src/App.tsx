import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import Layout from "./components/Layout";
import To_do from "./pages/Home";

function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Layout />}>
          <Route path="/" element={<To_do />} />
          
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default App;
