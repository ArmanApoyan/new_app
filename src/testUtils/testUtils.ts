import { combineReducers, createStore } from "redux";
import { myReducer } from "../store/Task/reducer";


export function createTestStore() {
    const store = createStore(
      combineReducers({
        task: myReducer,
      })
    );
    return store;
  }