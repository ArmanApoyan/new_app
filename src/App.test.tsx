import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import App from "./App";
import { createTestStore } from "./testUtils/testUtils";

describe("App test", () => {
  const store = createTestStore();

  test("renders learn react link", () => {
    const { container } = render(
        <Provider store={store}>
          <App />
        </Provider>
    );
    expect(container).toMatchSnapshot()
  });
});
