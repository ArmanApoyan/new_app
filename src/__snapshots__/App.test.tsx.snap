// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test renders learn react link 1`] = `
<div>
  <nav>
    <div
      class="head"
    >
      <div
        class="search"
      >
        <form
          class="searchForm"
        >
          <input
            class="search"
            placeholder="Search"
            type="search"
            value=""
          />
          <svg
            class="searchIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.742 10.344a6.5 6.5 0 1 0-1.397 1.398h-.001c.03.04.062.078.098.115l3.85 3.85a1 1 0 0 0 1.415-1.414l-3.85-3.85a1.007 1.007 0 0 0-.115-.1zM12 6.5a5.5 5.5 0 1 1-11 0 5.5 5.5 0 0 1 11 0z"
            />
          </svg>
        </form>
      </div>
      <div
        class="buttons"
      >
        <button
          class="add"
          data-testid="add"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M480 580H372a8 8 0 0 0-8 8v48a8 8 0 0 0 8 8h108v108a8 8 0 0 0 8 8h48a8 8 0 0 0 8-8V644h108a8 8 0 0 0 8-8v-48a8 8 0 0 0-8-8H544V472a8 8 0 0 0-8-8h-48a8 8 0 0 0-8 8v108zm374.6-291.3c6 6 9.4 14.1 9.4 22.6V928c0 17.7-14.3 32-32 32H192c-17.7 0-32-14.3-32-32V96c0-17.7 14.3-32 32-32h424.7c8.5 0 16.7 3.4 22.7 9.4l215.2 215.3zM790.2 326L602 137.8V326h188.2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </nav>
  <main
    class="main"
  >
    <div
      class="columns"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="To Do"
      data-testid="column"
    >
      <h2>
        To Do
      </h2>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="goal"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="1"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="1"
        data-testid="task"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <h1
          data-testid="title"
        >
          Task 1
        </h1>
        <h3>
          Description 1
        </h3>
        <button
          class="x"
          data-testid="close"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
        <button
          class="edit"
          data-testid="edit_btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="columns"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="On Progress"
      data-testid="column"
    >
      <h2>
        On Progress
      </h2>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="goal"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="2"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="2"
        data-testid="task"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <h1
          data-testid="title"
        >
          Task 2
        </h1>
        <h3>
          Description 2
        </h3>
        <button
          class="x"
          data-testid="close"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
        <button
          class="edit"
          data-testid="edit_btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="columns"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="Done"
      data-testid="column"
    >
      <h2>
        Done
      </h2>
      <div
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        class="goal"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="3"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="3"
        data-testid="task"
        draggable="false"
        role="button"
        tabindex="0"
      >
        <h1
          data-testid="title"
        >
          Task 3
        </h1>
        <h3>
          Description 3
        </h3>
        <button
          class="x"
          data-testid="close"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </button>
        <button
          class="edit"
          data-testid="edit_btn"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z"
            />
          </svg>
        </button>
      </div>
    </div>
  </main>
</div>
`;
